### PR TITLE
add modal before replacing metrics

### PR DESCRIPTION
**Pull request by bcolloran**

### CODE REVIEW

**Comment by bcolloran** [comment]
anyone can review this code, maybe @djbarnwal  since you've worked on the modal. also requesting review from @magorlick for strings

**Review by magorlick — APPROVED** [review]
The strings "continue" and "cancel" for the CTAs are good.

File: src/lib/components/metrics-definition/MetricsDefinitionGenerateButton.svelte [hunk 1]
```diff
@@ -25,12 +27,29 @@
     tooltipText = undefined;
     buttonDisabled = false;
   }
+
+  let modalIsOpen = false;
+
+  const openModal = () => {
+    console.log("opening");
```
Comment (djbarnwal) [inline src/lib/components/metrics-definition/MetricsDefinitionGenerateButton.svelte:34]: Can we remove this log?

File: src/lib/components/metrics-definition/MetricsDefinitionGenerateButton.svelte [hunk 2]
```diff
@@ -25,12 +27,29 @@
...
+    modalIsOpen = true;
+  };
+
+  const closeModal = () => {
+    console.log("opening");
```
Comment (djbarnwal) [inline src/lib/components/metrics-definition/MetricsDefinitionGenerateButton.svelte:39]: This as well

**Review by djbarnwal — CHANGES_REQUESTED** [review]

File: src/lib/components/metrics-definition/MetricsDefinitionGenerateButton.svelte (hunk 1 above)
Comment (bcolloran) [reply src/lib/components/metrics-definition/MetricsDefinitionGenerateButton.svelte:34]: oops, yes, thanks for catching these @djbarnwal ! :-)

**Pull request merged**